### PR TITLE
Players/Dungeons&Raids: Respawn alive at graveyard

### PR DESCRIPTION
**Pull request by Golrag**
**Changes proposed:**

-  Allow Players to respawn alive in dungeons & raids

**Issues addressed:**

Closes: none?

**Tests performed:**

- Builds
- Died in underrot, respawned alive

**Known issues and TODO list:**

- [ ] There is a flag NoGhostOnRelease, im not sure why this one is not set for underrot areas.

SQL:

```sql
INSERT INTO `graveyard_zone` (`ID`, `GhostZone`, `Faction`, `Comment`) VALUES
(6378, 9391, 0, 'Underrot Graveyard');
```

### CODE REVIEW

**Comment by Nyr97** [comment]
Can anyone confirm if old vanilla dungeons or raids currently skip ghost stuff if you joined without LFG currently?

**Comment by Golrag** [comment]
In ragefire chasm you do

**Comment by Aokromes** [comment]
and on shadowfang keep, i know it was revamped, but, you resurrect inside after die.

**Comment by Nyr97** [comment]
It's been so long since I walked into a dungeon on retail... but yeah, apparently the ghost-trip has been over for a while after dying.

File: src/server/game/Entities/Player/Player.cpp [hunk 1]
```diff
@@ -4831,7 +4831,7 @@ void Player::RepopAtGraveyard()
 
     bool shouldResurrect = false;
     // Such zones are considered unreachable as a ghost and the player must be automatically revived
-    if ((!IsAlive() && zone && zone->GetFlags().HasFlag(AreaFlags::NoGhostOnRelease)) || GetTransport() || GetPositionZ() < GetMap()->GetMinHeight(GetPhaseShift(), GetPositionX(), GetPositionY()))
+    if ((!IsAlive() && zone && zone->GetFlags().HasFlag(AreaFlags::NoGhostOnRelease)) || GetMap()->IsNonRaidDungeon() || GetMap()->IsRaid() || GetTransport() || GetPositionZ() < GetMap()->GetMinHeight(GetPhaseShift(), GetPositionX(), GetPositionY()))
```
Comment (Shauren) [inline src/server/game/Entities/Player/Player.cpp:4834]: I think this should be just `GetMap()->IsDungeon()` (covers dungeon, raid and scenario)

Comment (Golrag) [reply src/server/game/Entities/Player/Player.cpp:4834]: There are scenarios that have spirit healers in mists of pandaria, its been too long and im not sure if we can queue for them in dragonflight

Comment (Shauren) [reply src/server/game/Entities/Player/Player.cpp:4834]: yes you can queue for them

Comment (Golrag) [reply src/server/game/Entities/Player/Player.cpp:4834]: just tested to make sure, yes you get ghost when you die. Maybe not in all of them but I think that might just be done with AreaFlags::NoGhostOnRelease

**Pull request merged**